### PR TITLE
Framework: remove sites list usage from media

### PR DESCRIPTION
**Pull request by gwwar**
This PR removes SitesList usage from the media section.

### Testing Instructions
- Navigate to http://calypso.localhost:3000/media
- Select a site
- You can: upload new images, filter, edit, restore, and modify meta data like titles
- No functional changes

### CODE REVIEW

File: client/my-sites/media/controller.js [hunk 1]
```diff
@@ -36,7 +43,8 @@ module.exports = {
 		// Render
 		renderWithReduxStore(
 			React.createElement( MediaComponent, {
-				sites: sites,
+				selectedSite,
+				selectedSiteSlug,
```
Comment (mtias) [inline client/my-sites/media/controller.js]: Can we just use `selectedSite.slug` inside the component and have one less prop? :)

Comment (gwwar) [inline client/my-sites/media/controller.js]: There's some corner cases for Jetpack sites that we should be using the selector for

Comment (gwwar) [inline client/my-sites/media/controller.js]: 👍  Updated in e5c1a5f

**Comment by jhnstn** [comment]
Test plan works for me :shipit:

**Comment by gwwar** [comment]
Thanks for the review @jhnstn and @mtias!
